uint8 id_old
uint8 id_new
